OptionSelector/AnimalTypesTableViewController.m

```diff
@@ -0,0 +1,178 @@
+                           @"Alaskan Malamute",
+                           @"Pug",
+                           @"Bulldog",
+                           @"Bernese Mountain Dog",
+                           @"Labrador Retriver",
+                           @"Mastiff"];
+    dog.selection = @" ";
+    
+    
+    
+    
+    
+    // breeds of cat - these are our options to choose from
+    
+    cat.options = @[
+                           @"Russian Blue",
+                           @"Calico",
+                           @"Siamese",
+                           @"British Shorthair",
+                           @"Persian",
+                           @"Cornish Rex",
+                           @"Scottish Fold"];
+    
+    cat.selection = @" ";
+    
+    
+    
+    
+    
+    // breeds of bug - these are our options to choose from
+    
+    bug.options = @[
+                           @"Butterfly",
+                           @"Ladybug",
+                           @"Caterpillar",
+                           @"Spider",
+                           @"Dragonfly",
+                           @"Cricket"];
+    
+    bug.selection = @" ";
+    
+    
+    
+    // adding these CQCategory instancse - dog, cat, bug - to our array to hold the 'types' of animal
+    
+    [self.array addObject:dog];
+    [self.array addObject:cat];
+    [self.array addObject:bug];
+    
+}
+
+- (void)viewWillAppear:(BOOL)animated{
+    [super viewWillAppear:animated];
+    [self.tableView reloadData];
+}
+- (void)didReceiveMemoryWarning {
+    [super didReceiveMemoryWarning];
+    // Dispose of any resources that can be recreated.
+}
+
+
+- (void)addItemViewController:(AnimalDetailTableViewController *)controller didFinishEnteringItem:(CQCategory *)item{
+
+    [self.array replaceObjectAtIndex:self.index withObject:item];
+    
+}
+
+
+
+
+#pragma mark - Table view data source
+
+- (NSInteger)numberOfSectionsInTableView:(UITableView *)tableView {
+    return 1;
+}
+
+- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
+    return self.array.count;
+}
+
+
+- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
+    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:@"AnimalTypesIdentifier" forIndexPath:indexPath];
+    
+    CQCategory *object = [self.array objectAtIndex:indexPath.row];
+    NSString *name = object.name;
+    
+    cell.textLabel.text = name;
+    cell.detailTextLabel.text = [object.selection isEqualToString:@" "]?@" ":object.selection;
+    return cell;
+}
+
+
+#pragma mark - Navigation
+
+
+- (void)prepareForSegue:(UIStoryboardSegue *)segue sender:(id)sender {
+    
+    NSLog(@"You clicked on an animal with a specific breed.");
+    
+    NSIndexPath *indexPath = [self.tableView indexPathForSelectedRow];
+    
+    
+    AnimalDetailTableViewController *detailViewController = segue.destinationViewController;
+    
+    self.index = indexPath.row;
```

Comment:
I like that you've cached the current index of the category selected. :+1: Makes it much easier to reference the category from the delegate method.

However, I think the property name could be more descriptive. When I read an variable name like `index`, I assume that it has something to do with an array, but without more context I don't know what that index is supposed to represent.

In this case, you might consider a variable name of (`indexOfSelectedCategory` or `currentAnimalIndex`). That way, when you see it used in your code, it's purpose is more clear.